api/QCVOC.Api/Service/Controller/ServiceController.cs

```diff
@@ -0,0 +1,214 @@
+        /// <returns>See attributes.</returns>
+        /// <response code="201">The Service was created successfully.</response>
+        /// <response code="400">The specified Service was invalid.</response>
+        /// <response code="401">Unauthorized.</response>
+        /// <response code="409">A Service with the same name already exists.</response>
+        /// <response code="500">The server encountered an error while processing the request.</response>
+        [HttpPost("")]
+        [Authorize]
+        [ProducesResponseType(typeof(Service), 201)]
+        [ProducesResponseType(typeof(ModelStateDictionary), 400)]
+        [ProducesResponseType(401)]
+        [ProducesResponseType(409)]
+        [ProducesResponseType(typeof(Exception), 500)]
+        public IActionResult Create([FromBody]ServiceAddRequest service)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var existingService = ServiceRepository.GetAll(new ServiceFilters()
+            {
+                Name = service.Name
+            });
+
+            if (existingService.Any())
+            {
+                return Conflict($"A Service with a matching name already exists.");
+            }
+
+            var serviceRecord = new Service()
+            {
+                Name = service.Name,
+                Description = service.Description,
+                Id = Guid.NewGuid(),
+                CreationDate = DateTime.UtcNow,
+                CreationById = User.GetId(),
+                LastUpdateDate = DateTime.UtcNow,
+                LastUpdateById = User.GetId(),
+            };
+
+            try
+            {
+                var createdService = ServiceRepository.Create(serviceRecord);
+                return StatusCode(201, createdService);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception($"Error creating the specified Service: {ex.Message}. See inner exception for details.", ex);
+            }
+        }
+
+        /// <summary>
+        ///     Updates the Service matching the specified <paramref name="id"/>.
+        /// </summary>
+        /// <param name="id">The id of the Service to update.</param>
+        /// <param name="service">The updated Service.</param>
+        /// <returns>See attributes.</returns>
+        /// <response code="200">The Service was updated successfully.</response>
+        /// <response code="400">The specified Service was invalid.</response>
+        /// <response code="401">Unauthorized.</response>
+        /// <response code="404">A Service matching the specified id could not be found.</response>
+        /// <response code="409">A Service with the same name already exists.</response>
+        /// <response code="500">The server encountered an error while processing the request.</response>
+        [HttpPut("{id}")]
+        [ProducesResponseType(typeof(Service), 200)]
```

Comment:
Let's restrict this to just admins and supervisors: `[Authorize(Roles = nameof(Role.Administrator) + "," + nameof(Role.Supervisor))]`

Reply:
Mea culpa